src/content/docs/reference/driver-guide.md

```diff
@@ -0,0 +1,452 @@
+      }
+    });
+  }
+
+  const headers = {
+    Authorization: `Bearer ${state.API_KEY}`,
+    "Content-Type": "application/json",
+  };
+
+  const response = await fetch(url.toString(), {
+    method,
+    headers,
+    body,
+  });
+
+  if (!response.ok) {
+    throw new Error(`API error: ${response.status} ${response.statusText}`);
+  }
+
+  // Note: Response handling varies by API. Some APIs:
+  // - Always return JSON
+  // - Mix JSON and non-JSON responses
+  // - Return different content types for errors vs success
+  // Adjust this logic based on your API's behavior
+  const contentType = response.headers.get("content-type");
+  if (contentType && contentType.includes("application/json")) {
+    return response.json();
+  } else {
+    return response.text();
+  }
+}
+
+```
+
+### Root Object
+
+In your driver’s `index.ts` file, the Root object serves as the entry point to your driver. It defines the top-level structure and available operations. For Resend, we expose email and domain operations through collection fields:
+
+```js
+// index.ts
+import { root } from "membrane";
+import { status, configure, api } from "./helpers";
+import { emailTests, domainTests } from "./tests";
+
+export const Root = {
+  // Status check for configuration
+  status,
+
+  // Configuration action
+  configure,
+
+  // Collection fields
+  emails: () => ({}),
+  domains: () => ({}),
+
+  // Test fields
+  tests: () => ({}),
+};
+```
+
+### Resources, Collections, and Pages
+
+In Membrane drivers, we organize our code around key resources.
+
+The pattern is to create pairs of Collection and Resource objects in the `index.ts` file, where collections handle operations like listing and creation, while resources handle item-specific operations.
+
+This Resource-Collection-Page pattern in Membrane provides a consistent interface where:
+
+- Collections handle `one()` for getting single resources, `page()` for listing resources, and other collection-level operations like `create()` or `search()`
+- Resources implement [`gref`](#graph-references-grefs) for referencing and item-specific operations like `update()` and `delete()`
+- Collection operations go on collections (listing, creating)
+- Resource-specific operations go on the resource object (updating, deleting)
+
+Let's look at this pattern using Resend's Emails and Domains as examples.
+
+### Email Collection
+
+Collections typically implement `one()` for getting single resources and methods for creating new resources:
+
+```js
+// index.ts
+export const EmailCollection = {
+  // Get a single email by ID
+  async one({ id }: { id: string }) {
+    const data = await api("GET", `emails/${id}`);
+    return { ...data };
+  },
+
+  // Send a new email
+  async send(args: {
+    from: string;
+    to: string[];
+    subject: string;
+    html?: string;
+    text?: string;
+  }) {
+    const data = await api("POST", "emails", undefined, args);
+    return { ...data };
+  }
+};
```

Comment:
Hm, do we need `send` here if the point is just to illustrate the `.one()` pattern?